Control/Monad/Error/Class.hs

```diff
@@ -206,6 +207,13 @@ instance
     throwError = lift . throwError
     catchError = Accum.liftCatch catchError
 
+instance (MonadError e m, MonadError e n) => MonadError e (Product m n) where
+    throwError e = Pair (throwError e) (throwError e)
+    catchError (Pair ma na) f = Pair (catchError ma (productFst . f)) (catchError na (productSnd. f))
```

Comment:
Can you please write `catchError` in infix notation, at least on the right-hand side?  It can reduce nested parentheses.  Also the final `.` needs space around it.

Reply:
> Can you please write catchError in infix notation, at least on the right-hand side?

When I do this:

```
Pair (ma `catchError` productFst . f) (na `catchError` productSnd . f)
```

I get a precedence parsing error (`catchError` is `infixl 9` and `.` is `infixr 9`).

I could use infix `Pair` to reduce parentheses:

```
catchError ma (productFst . f) `Pair` catchError na (productSnd . f)
```

Thoughts?

Reply:
> Also the final . needs space around it.

Fixed in 22c80e7

Reply:
> Thoughts?

My thoughts are that my cellphone is not the best tool for collaborating on GitHub issues.

Thanks for the contribution!